django/db/models/base.py

```diff
@@ -370,15 +370,22 @@ def _default_manager(cls):
         return cls._meta.default_manager
 
 
+class ModelStateFieldsCacheDescriptor:
+    def __get__(self, instance, cls):
+        if instance is None:
+            return self
```

Comment:
@sir-sigurd it's needed to prevent accesses to `ModelState.fields_cache` from crashing which is something some IDEs do under the hood for introspection. See dc1ad69f30f48d3c27406373f62f87e93550cdc5 and b88abd684041ffa66bfe445e1ac26164e803d488.